OsmAnd/src/net/osmand/plus/plugins/odb/OBDWidgetSettingFragment.kt

```diff
@@ -81,18 +88,29 @@ class AverageModeSettingFragment : BaseSimpleWidgetSettingsFragment() {
 	private fun setupConfigButtons() {
 		buttonsCard?.removeAllViews()
 
-		buttonsCard?.addView(createButtonWithDescription(
-			getString(R.string.shared_string_mode),
-			getModeName(selectedAverageMode),
-			selectedAverageMode,
-		) { showMarkerModeDialog() })
+		val showAverageButtons = averageValueModePref != null && averageValueIntervalPref != null
+		if (temperatureUnitPref != null) {
+			buttonsCard?.addView(createButtonWithDescription(
+				getString(R.string.shared_string_temperature),
+				app.getString(selectedTemperatureUnit.titleId),
+				showAverageButtons,
+			) { showTemperatureDialog() })
+		}
 
-		if (selectedAverageMode) {
+		if (showAverageButtons) {
 			buttonsCard?.addView(createButtonWithDescription(
-				getString(R.string.shared_string_interval),
-				availableIntervals[selectedIntervalMillis],
-				false,
-			) { showSeekbarSettingsDialog() })
+				getString(R.string.shared_string_mode),
+				getModeName(selectedAverageMode),
+				selectedAverageMode,
+			) { showMarkerModeDialog() })
+
+			if (selectedAverageMode) {
+				buttonsCard?.addView(createButtonWithDescription(
+					getString(R.string.shared_string_interval),
+					availableIntervals[selectedIntervalMillis],
+					false,
+				) { showSeekbarSettingsDialog() })
+			}
```

Comment:
It looks complicated, can we simplify it? Because adding new parameters will be harder...
 
What if we use list  - inner class ButtonItem(title: String, desc: String, listener: View.OnClickListener), showShortDivider will be for last item